src/sage/graphs/graph_decompositions/slice_decomposition.pyx

```diff
@@ -0,0 +1,970 @@
+                sigma[i] = v_int
+                deref(sigma_inv)[v_int] = i
+                i = i + 1
+            max_degree = max(max_degree, cg.out_degrees[v_int])
+
+    # Variables needed to iterate over neighbors of a vertex
+    cdef int nneighbors
+    cdef vector[int] neighbors = vector[int](max_degree)
+
+    # Initialize partition: one part containing all the vertices
+    nparts = 1
+    # all element of part_of are already initialized to 0
+    part_head[0] = 0
+    subpart[0] = 0
+    if xslice_len != NULL:
+        part_len[0] = n
+
+    # Main loop
+    for i in range(n):
+        old_nparts = nparts
+
+        part_of_i = part_of[i]
+
+        # put i out of its part (updating part_len if needed)
+        part_head[part_of_i] += 1
+        if xslice_len != NULL:
+            deref(xslice_len)[i] = part_len[part_of_i]
+            part_len[part_of_i] -= 1
+
+        v_int = sigma[i]
+
+        # Iterate over the neighbors of v
+        nneighbors = cg.out_neighbors_unsafe (v_int, neighbors.data(), max_degree)
+        for k in range(nneighbors):
+            u_int = neighbors[k]
+            j = deref(sigma_inv)[u_int] # get the position of u
+            if j <= i:
+                continue # already taken care of
+
+            if lex_label != NULL:
+                deref(lex_label)[j].push_back (v_int)
+
+            p = part_of[j] # get the part of u
+            l = part_head[p] # get the beginning of the part containing u
+
+            # if not last and next elem belongs in the same part (ie #part >= 2)
+            if l < n - 1 and part_of[l + 1] == p:
+                if l != j: # not already first elem of the part
+                    # Place u at the position of the head of the part
+                    t_int = sigma[l]
+                    deref(sigma_inv)[t_int], deref(sigma_inv)[u_int] = j, l
+                    sigma[j], sigma[l] = t_int, u_int
+                    if lex_label != NULL:
+                        swap[vector[int]](deref(lex_label)[j],
+                                          deref(lex_label)[l])
+                    j = l
+                part_head[p] += 1 # move the head of the part to next elem
+
+            # if part p was not already cut in two during this iteration, we
+            # create a new part using subpart
+            if subpart[p] < old_nparts:
+                subpart[p] = nparts
+                part_head[nparts] = j
+                if xslice_len != NULL:
+                    part_len[nparts] = 0
+                subpart[nparts] = 0
+                nparts += 1
+
+            # Finally, we update the name of the part for position j and set v
+            # as predecessor of u
+            part_of[j] = subpart[p]
+            if xslice_len != NULL:
+                part_len[p] -= 1
+                part_len[subpart[p]] += 1
+            if pred != NULL:
+                deref(pred)[j] = i
+
+    if need_to_delete_sigma_inv:
+        del sigma_inv
+
```

Comment:
2 empty lines between methods of a module. only 1 empty line between methods of a class.

Reply:
done in 8ffab789b46d677789f96faee3917ef439c667a3